sdk/storage/azure-storage-file-share/azure/storage/fileshare/_download.py

```diff
@@ -162,6 +163,9 @@ def _download_chunk(self, chunk_start, chunk_end):
                 download_stream_current=self.progress_total,
                 **self.request_options
             )
+            if response.properties.etag != self.etag:
+                raise ValueError("The file has been modified while downloading.")
```

Comment:
Please check with @annatisch if this is right exception type.

Reply:
In dotnet we throw [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareFileModifiedException.cs)  and provide more data than just message.

Reply:
Confirmed with @xiafu-msft and we are now throwing `ResourceModifiedError` as per blob natively does in this case.